scala3/src/main/scala/training/Syntax.scala

```diff
@@ -0,0 +1,20 @@
+package training
+import training.DataTypes.*
+import training.Typeclases.*
+import training.Instances.given
+
+object Syntax:
+  extension [A: Combinator](aMaybe: Maybe[A])
+    def +(bMaybe: Maybe[A]): Maybe[A] = (aMaybe, bMaybe) match
+      case (No, No)         => No
+      case (Yes(_), No)     => aMaybe
+      case (No, Yes(_))     => bMaybe
+      case (Yes(a), Yes(b)) => Yes(summon[Combinator[A]].combine(a, b))
+
+  extension [A](self: Maybe[A])
+    def map[B](f: A => B): Maybe[B] = summon[Transformer[Maybe]].map(self, f)
+
+  extension [A](self: A) def pure[F[_]](using sv: Lifter[F]): F[A] = sv.pure(self)
```

Comment:
@rafaparadela this has been annoying me. I really want to describe something like this 
```scala
extension [A: Lifter[F]](self: A) def pure [F[_]]: F[A] = summon[Lifter[F]].pure(self)
```
 but I am not able to describe it well and get compiler errors. And at this point, I am just doing brute force. 🤣

Reply:
Solved it, this is the use case for context functions `?=>`